components/identities/actions/identityActions.tsx

```diff
@@ -66,54 +69,75 @@ const IdentityActions: FunctionComponent<IdentityActionsProps> = ({
 
   const nextAutoRenew = useMemo(() => {
     const now = Math.floor(Date.now() / 1000);
-    if (identity?.domain_expiry) {
-      if (identity?.domain_expiry + 2592000 < now) {
+    if (identity?.getDomainExpiry()) {
+      if ((identity?.getDomainExpiry() as number) + 2592000 < now) {
         return "Next today";
       } else {
         return (
           "Next on " +
-          timestampToReadableDate(identity?.domain_expiry - 2592000)
+          timestampToReadableDate(
+            (identity?.getDomainExpiry() as number) - 2592000
+          )
         );
       }
     }
-  }, [identity?.domain_expiry]);
+  }, [identity]);
 
   // Add all subdomains to the parameters
   const callDataEncodedDomain: (number | string)[] = [encodedDomains.length];
   encodedDomains.forEach((domain) => {
     callDataEncodedDomain.push(domain.toString(10));
   });
 
-  //Set as main domain execute
-  const set_address_to_domain_calls = {
+  // todo: optimize by skipping this call if the user doesn't have an old main domain
+  const resetAddrToDomain = {
     contractAddress: process.env.NEXT_PUBLIC_NAMING_CONTRACT as string,
-    entrypoint: "set_address_to_domain",
-    calldata: callDataEncodedDomain,
+    entrypoint: "reset_address_to_domain",
+    calldata: [],
   };
-  const set_domain_to_address_calls = {
+
+  const resetLegacyDomainToAddr = {
     contractAddress: process.env.NEXT_PUBLIC_NAMING_CONTRACT as string,
-    entrypoint: "set_domain_to_address",
-    calldata: [...callDataEncodedDomain, hexToDecimal(address)],
+    entrypoint: "clear_legacy_domain_to_address",
+    calldata: callDataEncodedDomain,
   };
-  const { writeAsync: set_address_to_domain, data: mainDomainData } =
-    useContractWrite({
-      calls: isAccountTargetAddress
-        ? [set_address_to_domain_calls]
-        : [set_domain_to_address_calls, set_address_to_domain_calls],
-    });
 
-  function setAddressToDomain(): void {
-    set_address_to_domain();
-  }
+  const resetDomainToAddr = {
```

Comment:
same here. 

Apart from that what is the difference between `resetDomainToAddr` and `resetDomainToAddr`

Reply:
![image](https://github.com/starknet-id/app.starknet.id/assets/41830259/f820ee1e-188d-40bd-83c0-3476e77c8837)

Reply:
Joke apart I really didn't get the question